### PR TITLE
Document color_mode and white for MQTT basic light

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Document `color_mode` and `white` support for MQTT basic light

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/51484

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/light.mqtt.markdown [hunk 1]
```diff
@@ -15,15 +15,17 @@ The `mqtt` light platform lets you control your MQTT enabled lights through one
...
+| RGBWW Color       | ✔                                                          | ✔                                                                    | ✘                                                                            |
 | Transitions       | ✘                                                          | ✔                                                                    | ✔                                                                            |
-| XY Color          | ✔                                                          | ✔                                                                    | ✘                                                                            |
-| HS Color          | ✔                                                          | ✔                                                                    | ✘                                                                            |
 | White Value       | ✔                                                          | ✔                                                                    | ✔                                                                            |
```
Comment (MartinHjelmare) [inline source/_integrations/light.mqtt.markdown]: Are we going to deprecate white value? Should we mention somewhere the relationship between white value and white?

Comment (emontnemery) [reply source/_integrations/light.mqtt.markdown]: We could just remove it instead of documenting deprecated features?

Comment (frenck) [reply source/_integrations/light.mqtt.markdown]: That is the general approach in end-user documentation. Removing deprecated documentation, which in general, works pretty well.

**Review by frenck — APPROVED** [review]
LGTM ✅

**Pull request merged**